internal/ingress/election/election.go

```diff
@@ -35,9 +36,31 @@ type elector struct {
 	elector *leaderelection.LeaderElector
 }
 
-func (e elector) Run() {
-	// TODO pass the correct context in here
-	e.elector.Run(context.Background())
+func (e elector) Run(ctx context.Context) {
+	backoff := flowcontrol.NewBackOff(1*time.Second, 15*time.Second)
+
+	for session := 0; ; {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			first := session == 0
+			session++
+			if !first {
+				const id = ""
```

Comment:
Can we please add a name here for the `id`?
I don't have a very good name myself but something along the lines of `kong-leader-election` would be fine.
Please move the definition of this outside the loop and at the method level itself.